cmd/operator/main.go

```diff
@@ -71,44 +140,19 @@ func main() {
 		LeaderElectionID:   "d1c7b6d5.intel.com",
 	})
 	if err != nil {
-		setupLog.Error(err, "unable to start manager")
+		ctrl.Log.WithName("setup").Error(err, "unable to start manager")
 		os.Exit(1)
 	}
 
-	if err = dsa.SetupReconciler(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "DsaDevicePlugin")
-		os.Exit(1)
-	}
-	if err = (&devicepluginv1.DsaDevicePlugin{}).SetupWebhookWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create webhook", "webhook", "DsaDevicePlugin")
-		os.Exit(1)
+	ns := os.Getenv("DEVICEPLUGIN_NAMESPACE")
+	if ns == "" {
+		ns = devicePluginNamespace
 	}
 
-	if err = gpu.SetupReconciler(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "GpuDevicePlugin")
-		os.Exit(1)
-	}
-	if err = (&devicepluginv1.GpuDevicePlugin{}).SetupWebhookWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create webhook", "webhook", "GpuDevicePlugin")
-		os.Exit(1)
-	}
-
-	if err = qat.SetupReconciler(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "QatDevicePlugin")
-		os.Exit(1)
-	}
-	if err = (&devicepluginv1.QatDevicePlugin{}).SetupWebhookWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create webhook", "webhook", "QatDevicePlugin")
-		os.Exit(1)
-	}
-
-	if err = sgx.SetupReconciler(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "SgxDevicePlugin")
-		os.Exit(1)
-	}
-	if err = (&devicepluginv1.SgxDevicePlugin{}).SetupWebhookWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create webhook", "webhook", "SgxDevicePlugin")
-		os.Exit(1)
+	// TODO(mythi): add a StringVar flag to select which controllers to enable
+	for _, device := range []string{"dsa", "fpga", "gpu", "qat", "sgx"} {
+		createDevicePluginController(mgr, ns, device)
+		createDevicePluginWebhook(mgr, device)
```

Comment:
I had to split these into two because one was still too "complex"

Reply:
Perhaps moving `*.SetupWebhookWithManager()` into `*.SetupReconsiler()` could also work. Then we could introduce a mapping of operator names to `*.SetupReconsiler` functions and call the functions inside the loop. This way the code would be much shorter:
```go
for _, device := range []string{"dsa", "fpga", "gpu", "qat", "sgx"} {
        if err := setupReconcilerAndWebhookFuncs[device](mgr, ns, device); err != nil {
                setupLog.Error(...)
                os.Exit(1)
        }
}
```

Reply:
@rojkov I submitted v2.